### PR TITLE
PR form solution

### PR DESCRIPTION
**Pull request by Jacobix95**
My solution for the assignment

### CODE REVIEW

File: src/test/java/utils/ConfigReader.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+public class ConfigReader {
+    private static final Properties properties = new Properties();
+
+    static {
+        try (InputStream input = ConfigReader.class.getClassLoader().getResourceAsStream("config.properties")) {
+            if (input != null) {
+                properties.load(input);
+            } else {
+                throw new RuntimeException("config.properties file not found in classpath.");
```
Comment (gavura256) [inline src/test/java/utils/ConfigReader.java]: RuntimeException is good but, IllegalArgumentException is better

File: src/test/java/tests/LoginTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Test
+    public void testValidCredentials_UC3() {
+        LoginPage login = new LoginPage(getDriver());
+        login.enterUsername("standard_user");
+        login.enterPassword("secret_sauce");
+        login.clickLogin();
+        Assertions.assertThat(getDriver().getCurrentUrl()).contains("inventory");
+    }
+}
```
Comment (gavura256) [inline src/test/java/tests/LoginTest.java]: Please add an empty line at the end of the class

File: src/test/java/tests/LoginTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import static utils.ErrorMessage.PASSWORD_REQUIRED;
+import static utils.ErrorMessage.USERNAME_REQUIRED;
+
+public class LoginTest extends BaseTest {
+
+    @Test
+    public void testEmptyCredentials_UC1() {
+        LoginPage login = new LoginPage(getDriver());
+        login.enterUsername("testUser");
```
Comment (gavura256) [inline src/test/java/tests/LoginTest.java]: Why we need to pass anything when want to verify test case with empty credentials?

File: src/test/java/runners/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    public WebDriver getDriver() {
+        return driver.get();
+    }
+
+    @BeforeMethod
+    public void setUp() {
+        String browser = ConfigReader.getBrowser();
+        driver.set(WebDriverFactory.getDriver(browser));
+        getDriver().get("https://www.saucedemo.com/");
```
Comment (gavura256) [inline src/test/java/runners/BaseTest.java]: I suggest you to extract the link into constant

Comment (gavura256) [reply src/test/java/runners/BaseTest.java]: Why are you opening the link in setup method?

File: src/test/java/tests/LoginTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        login.clearPassword();
+        login.clickLogin();
+        Assertions.assertThat(login.getErrorMessage()).isEqualTo(USERNAME_REQUIRED);
+    }
+
+    @Test
+    public void testUsernameOnly_UC2() {
+        LoginPage login = new LoginPage(getDriver());
+        login.enterUsername("standard_user");
+        login.enterPassword("anyPass");
```
Comment (gavura256) [inline src/test/java/tests/LoginTest.java]: Why do you pass password and after that remove it? Can we skip these steps

File: src/test/resources/config.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+browser=chrome
```
Comment (gavura256) [inline src/test/resources/config.properties]: I'm suggesting you to take a look how to passing the browser through environment variables. No need to rework this, just be prepared to explain to me how to do that during 1-2-1 session

Comment (gavura256) [reply src/test/resources/config.properties]: and add please blank line at the end of the file

**Review by gavura256 — CHANGES_REQUESTED** [review]

File: src/test/java/tests/LoginTest.java (hunk 3 above)
Comment (Jacobix95) [reply src/test/java/tests/LoginTest.java]: I will try to explain it in 1-2-1 session

File: src/test/java/tests/LoginTest.java (hunk 5 above)
Comment (Jacobix95) [reply src/test/java/tests/LoginTest.java]: I will try to explain it in 1-2-1 session
